discovery_test.go

```diff
@@ -0,0 +1,131 @@
+package oidc
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"reflect"
+	"testing"
+)
+
+type roundTripperFunc func(req *http.Request) (*http.Response, error)
+
+func (f roundTripperFunc) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f(req)
+}
+
+func Test_discover(t *testing.T) {
```

Comment:
Would you like us to standardize on Test_ as naming for the tests? I think we have used names without underscore in other places, but using underscore makes it more readable especially in cases like this where the function starts with a lowercase letter.

Reply:
Good point, I actually did not give this a lot of thought. I let Goland generate the test stubs and it gave me an underscore and I did not consider the other tests much.

Maybe we can give this subject some more thought and look at what other projects are doing and see what we like.